File: Windows/CodexDuo.Windows/CodexAuthCommands.cs
namespace CodexDuo.Windows;

public static class CodexAuthCommands
{
    public static IReadOnlyList<string> SwitchAccount(string accountKey) => ["switch", accountKey];
    public static IReadOnlyList<string> SetAlias(string accountKey, string? alias) => string.IsNullOrWhiteSpace(alias)
        ? ["alias", "clear", accountKey]
        : ["alias", "set", accountKey, alias.Trim()];
    public static IReadOnlyList<string> RemoveAccount(string accountKey) => ["remove", accountKey];
}

File: Sources/CodexDuo/CodexAuthCommands.swift
import Foundation

enum CodexAuthCommands {
    static func switchAccount(selector: String) -> [String] {
        ["switch", selector]
    }

    static func setAlias(selector: String, alias: String?) -> [String] {
        if let alias, !alias.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            return ["alias", "set", selector, alias]
        }
        return ["alias", "clear", selector]
    }

    static func removeAccount(selector: String) -> [String] {
        ["remove", selector]
    }
}
